### ISSUE TITLE
External SQL files in referenced libraries does not work if not copied to the referencing library

### ISSUE THREAD

**Issue by davidpodhola** [fsprojects/FSharp.Data.SqlClient#123]
if selecting from a table with column named [From], SQL error is reported, typecheck throws an error

**Comment by davidpodhola** [comment]
typecheck error FS3033: error : Incorrect syntax near 'SQL'.The batch could not be analyzed because of compile errors.

**Comment by dsevastianov** [comment]
Could you provide actual SQL as well

**Comment by davidpodhola** [comment]
Yep, now I need to get the "outer" code running, then I want to dig deeper into this (in about 8 hours I think)

**Comment by davidpodhola** [comment]
Just tried quickly getting the code out and it is getting worse :cry: 
https://gist.github.com/davidpodhola/025f19a5adc92898742e is working correctly so it must be related to some other type in the solution. 

Any suggestions how to debug it?

**Comment by davidpodhola** [comment]
removing the code to a separate project solved the issue. I would be still interested in how to get more information from the compilation. The issue also might be related to the type of the project, referenced projects etc.

**Comment by davidpodhola** [comment]
Generally something very bad happening in my fsproj :sad:
![capture](https://cloud.githubusercontent.com/assets/436605/6913923/eb02f254-d785-11e4-98a0-5cf0e0c2afd8.PNG)

**Comment by dmitry-a-morozov** [comment]
Do you use inline T-SQL or external *.sql files?

**Comment by davidpodhola** [comment]
I use external *.sql files. Added http://fsprojects.github.io/FSharp.Data.SqlClient/ to a project right now and I am unable to build it.
The mysterious `unknown(1,1)` error again:

```  C:\TFS\Sync.Today\Private\2015\sync.today\sync.today.io.flores\unknown(1,1):
error FS3033: The type provider 'FSharp.Data.SqlCommandProvider' reported an er
ror: Incorrect syntax near 'SQL'.↔The batch could not be analyzed because of co
mpile errors. [C:\TFS\Sync.Today\Private\2015\sync.today\sync.today.io.flores\s
ync.today.io.flores.fsproj]```

**Comment by davidpodhola** [comment]
Probably no way related to [From] since the actual SQL is 

```
select * from [dbo].[FloresActivities] where IsNew = 1
or InternalId not in ( select InternalId from  Appointments)
```

**Comment by davidpodhola** [comment]
So finally it is http://stackoverflow.com/questions/25156440/typecheck-error-fs3033-from-f-xmlprovider issue. If I copy the external files to the referencing library, it works.
Still not sure, how to use http://fsprojects.github.io/FSharp.Data.SqlClient/ in a bigger project, should the SQLs be referenced by absolute paths?

**Comment by dmitry-a-morozov** [comment]
Using file path with erased type that spawn more than one project is tricky,
If you have erased type defined in project A but use it in project B
actually code generation takes place in project B therefore file path (absolute or relative) has to be valid in context of project B too. 
This is a little confusing but that how erased type providers work by design.

**Comment by davidpodhola** [comment]
thanks @dmitry-a-morozov , you are right I forgot about how these works.
Anyway these types should not be propagated out of project A so this is the root issue in our case we need to fix.
Still, maybe somehow this could be mentioned somewhere in doc in http://fsprojects.github.io/FSharp.Data.SqlClient/configuration%20and%20input.html

**Comment by dmitry-a-morozov** [comment]
Sure. I will update the docs. 
This is not a first time this issue pops up.

**Comment by jon49** [comment]
I think I might have a similar issue to this one.

> The module/namespace 'FSharp.Data.SqlProgrammabilityProvider,ConnectionStringOrName="Server=... ;Database=FieldOps;User Id=...;Password=...".dbo.GetDSRContent' from compilation unit 'FSharp.Data.SqlClient' did not contain the namespace, module or type 'Record'

I add this line of code and it fails with the above error. This line of code looks exactly like 15
other functions I have except it calls a different stored procedure. Not sure if this deserves its
own issue or not. It is called in a separate project that is also F#, but the main project is C#,
but it is never used in the C# project. If I create a record and map it to the record I create no
error occurs.

```fsharp
    let getDSRReport reportID =
        use cmd = new WebServices.FieldOpsDb.dbo.GetDSRContent (Configuration.FieldOpsConnectionString)
        cmd.ExecuteSingle reportID
```

**Comment by dmitry-a-morozov** [comment]
Do you use *.sql file as command text?

**Comment by jon49** [comment]
The `SqlProgrammabilityProvider` is defined as

```fsharp
type FieldOpsDb = SqlProgrammabilityProvider<Configuration.FieldOpsConnectionStringLiteral>
```

I don't use *.sql for any commands for `SqlCommandProvider`. I tried, but couldn't get it to work and figured I would go back later. But the problem only occurs when using `SqlProgrammabilityProvider`.

**Comment by jon49** [comment]
This is what I'm using for the sql stored procedure:

```sql
ALTER PROCEDURE [dbo].[GetDSRContent] 
	-- Add the parameters for the stored procedure here
	@dealerServiceReportID int
AS
BEGIN
	-- SET NOCOUNT ON added to prevent extra result sets from
	-- interfering with SELECT statements.
	SET NOCOUNT ON;
	DECLARE
		@photoIDs varchar(200),
		@chartID int,
		@content nvarchar(max)

	SELECT @photoIDs = COALESCE(@photoIDs + ',', '') + CONVERT(varchar(20), p.DealerServiceReportPhotoID)
	FROM dbo.DealerServiceReportPhoto p
	WHERE p.DealerServiceReportID = @dealerServiceReportID

	SELECT @chartID = c.DealerServiceReportChartID
	FROM dbo.DealerServiceReportChart c
	WHERE c.DealerServiceReportID = @dealerServiceReportID


	SELECT @content = Content
	FROM dbo.DealerServiceReportContent d
	WHERE d.DealerServiceReportID = @dealerServiceReportID

	SELECT @content Content, @chartID ChartID, @photoIDs PhotoIDs

END
```

**Comment by vasily-kirichenko** [comment]
`SqlFile` TP is released in 1.8.4:

```fsharp
type F1 = SqlFile<"1.sql">
type Cmd2 = SqlCommandProvider<F1.Text, Cmd.str>
```
